nautobot/utilities/views.py

```diff
@@ -151,3 +187,1051 @@ def get_return_url(self, request, obj=None):
+
+        # Construct the objects table
+        table = self.table(self.queryset, user=request.user)
+        if "pk" in table.base_columns and (permissions["change"] or permissions["delete"]):
+            table.columns.show("pk")
+
+        # Apply the request context
+        paginate = {
+            "paginator_class": EnhancedPaginator,
+            "per_page": get_paginate_count(request),
+        }
+        RequestConfig(request, paginate).configure(table)
+
+        valid_actions = self.validate_action_buttons(request)
+
+        context = {
+            "content_type": content_type,
+            "table": table,
+            "permissions": permissions,
+            "action_buttons": valid_actions,
+            "table_config_form": TableConfigForm(table=table),
+            "filter_form": self.filterset_form(request.GET, label_suffix="") if self.filterset_form else None,
+        }
+        context.update(self.get_extra_context(request, "list", instance=None))
+
+        return render(request, self.template_name, context)
+
+    def alter_queryset_for_list(self, request):
+        # .all() is necessary to avoid caching queries
+        return self.queryset.all()
+
+
+class ObjectEditViewMixin(GetReturnURLMixin, NautobotViewSetMixin):
+    """
+    Create or edit a single object.
+    queryset: The base queryset for the object being modified
+    model_form: The form used to create or edit the object
+    template_name: The name of the template
+    """
+
+    def define_routes(self):
+        return super().define_routes() + [
+            Route(
+                url=r"^{prefix}/add/$",
+                mapping={
+                    "get": "handle_object_edit_get",
+                    "post": "handle_object_edit_post",
+                },
+                name="{basename}_add",
+                detail=False,
+                initkwargs={"suffix": "Add"},
+            ),
+            Route(
+                url=r"^{prefix}/{lookup}/edit/$",
+                mapping={
+                    "get": "handle_object_edit_get",
+                    "post": "handle_object_edit_post",
+                },
+                name="{basename}_edit",
+                detail=True,
+                initkwargs={"suffix": "Edit"},
+            ),
+        ]
+
+    def get_object_for_edit(self, kwargs):
+        # Look up an existing object by slug or PK, if provided.
+        if "slug" in kwargs:
+            return get_object_or_404(self.queryset, slug=kwargs["slug"])
+        elif "pk" in kwargs:
+            return get_object_or_404(self.queryset, pk=kwargs["pk"])
+        # Otherwise, return a new instance.
+        return self.queryset.model()
+
+    def alter_obj_for_edit(self, obj, request, url_args, url_kwargs):
+        # Allow views to add extra info to an object before it is processed. For example, a parent object can be defined
+        # given some parameter from the request URL.
+        return obj
```

Comment:
I am not clear on the difference between these? Why would you ever want to get an object for edit but not also alter it? Can we unify all of these under `get_object()` which is a DRF built-in method on model viewsets?